src/python/pants/jvm/jdk_rules.py

```diff
@@ -320,5 +334,43 @@ async def jvm_process(bash: BashBinary, request: JvmProcess) -> Process:
     )
 
 
+@rule
+async def jdk_request_for_target(ct: CoarsenedTarget) -> JdkRequest_:
+
+    # TODO: verify that we're requesting the same JDK version for all `ct` members?
+    t = ct.representative
```

Comment:
Yea. This validation could maybe move into `ClasspathEntryRequest`, and then become a field? Not a blocker though.

Reply:
@stuhood I think the validation side of things will require `ClasspathEntryRequest.for_targets` to be re-written as a rule (or further complicated by accepting more singleton arguments) to resolve the JDK at construction time. If you're OK with that, I can do that in a separate PR before landing this?